src/authenticator/formAuthenticator.js

```diff
@@ -88,7 +89,40 @@ FormAuthenticator.prototype.get = function(url){
   this._getField(this.passFieldSelector).sendKeys(this.pass);
   this._getField(this.logonButtonSelector).click().then(function () {
     // wait for all login actions to complete
-    that.statisticsCollector.authDone();
+    var stopProcessing = false;
+
+    function _pressAuthorizationButtonSelector() {
+      return function () {
+        if (stopProcessing) {
+          that.logger.debug("Stop waiting for authorizationButtonSelector.");
+          return true;
+        }
+        that._getField(that.authorizationButtonSelector).isEnabled().then(function (enabled) {
+          that.logger.debug("Is authorizationButtonSelector enabled: (" + enabled + ")");
+          if (enabled) {
+            //click Authorize button only after it is enabled
+            that._getField(that.authorizationButtonSelector).click().then(function () {
```

Comment:
just as a hint, you could extract the action. first _wait for the button to become enabled, and then click (control flow will take care of the sequence)